### PR TITLE
User can book a trip.

### PR DESCRIPTION
**Pull request by Robert-Jung**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -102,31 +111,158 @@ function createDetailPage(travelPackage) {
   return $div
 }
 
-//return to home
+//calculate total
```
Comment (thebearingedge) [inline main.js]: You can delete comments like this. Your function name is enough 😄

File: main.js [hunk 2]
```diff
@@ -24,9 +26,15 @@ var $nav = createNavbar()
 $nav.classList.add('navbar', 'navbar-inverse')
```
Comment (thebearingedge) [inline main.js]: Your `$nav` should be ready before `createNavbar` returns. Add the classes in the function that creates it!

File: main.js [hunk 3]
```diff
@@ -102,31 +111,158 @@ function createDetailPage(travelPackage) {
...
+  }
+  return total
+}
+
+//display total
+function displayTotal() {
+  var $div = document.getElementById('viewtotal')
+  var $span = document.createElement('span')
+
+  $div.innerHTML = ""
```
Comment (thebearingedge) [inline main.js]: Display total should take arguments. Calculate the total and pass it into `displayTotal`.

File: main.js [hunk 4]
```diff
@@ -102,31 +111,158 @@ function createDetailPage(travelPackage) {
...
+
+  $div.appendChild($img)
+  $div.appendChild($location)
+  $div.appendChild($price)
+  $div.appendChild($date)
+
+  return $div
+}
+
+//render checkout product
```
Comment (thebearingedge) [inline main.js]: You can delete comments like this. The function name is enough!

File: main.js [hunk 5]
```diff
@@ -102,31 +111,158 @@ function createDetailPage(travelPackage) {
...
+function renderCheckoutProduct() {
```
Comment (thebearingedge) [inline main.js]: Try to keep functions pure when possible, this function should return something instead of updating the DOM.

File: main.js [hunk 6]
```diff
@@ -102,31 +111,158 @@ function createDetailPage(travelPackage) {
...
+  var $viewCheckoutProduct = document.getElementById('productCheckout')
+  $viewCheckoutProduct.innerHTML = ""
+
+  for (var i = 0; i < app.cart.length; i++) {
+    var checkoutPackage = createCheckoutProduct(app.cart[i])
+    $viewCheckoutProduct.appendChild(checkoutPackage)
+  }
+}
+
+//views
```
Comment (thebearingedge) [inline main.js]: Try to refactor these functions into one general view-swapping strategy.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by Robert-Jung** [comment]
Hi Tim,
The requested changes are complete.

**Pull request merged**